### PR TITLE
Improve posterior KDEs for compressed data

### PR DESCRIPTION
**Pull request by lukashergt**
This PR addresses point 1 of issue #180.

The current version of anesthetic compresses the data before passing it to `kde=gaussian_kde(...)` to save time.  However, `gaussian_kde` is not actually the time consuming bit. Only the later _call_ to `kde(...)` is very time consuming. It is better to pass the full (uncompressed) dataset to `kde=gaussian_kde(...)` and then call `kde(...)` only on the compressed data. This won't take much longer, but will render much more precise posteriors.

# Checklist:

- [ ] I have performed a self-review of my own code
- [x] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [x] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [x] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by lukashergt** [comment]
The failing check from the test added in commit No. [2b554e5](https://github.com/williamjameshandley/anesthetic/pull/181/commits/2b554e57577263a9776fc0106d2dcf0cced8cad7) passes with the changes from commit No. [c81f474](https://github.com/williamjameshandley/anesthetic/pull/181/commits/c81f474a4a30867701d29ee7faf6a72edd3a6692), which demonstrates the improved precision we get from using the uncompressed data for `gaussian_kde`.

File: anesthetic/plot.py [hunk 1]
```diff
@@ -816,16 +816,10 @@ def kde_contour_plot_2d(ax, data_x, data_y, *args, **kwargs):
...
     cov = np.cov(data_x, data_y, aweights=weights)
     tri, w = triangular_sample_compression_2d(data_x, data_y, cov,
                                               weights, ncompress)
-    kde = gaussian_kde([tri.x, tri.y], weights=w)
-
-    x, y = kde.resample(ncompress)
-    x = np.concatenate([tri.x, x])
-    y = np.concatenate([tri.y, y])
-    w = np.concatenate([w, np.zeros(ncompress)])
-    tri = scaled_triangulation(x, y, cov)
```
Comment (lukashergt) [inline anesthetic/plot.py:828]: Was it ok to remove these lines? Tests don't complain...

File: tests/test_plot.py [hunk 2]
```diff
@@ -330,6 +330,17 @@ def test_kde_plot_1d(plot_1d):
...
 
+def test_ncompress_kde_1d():
+    fig, ax = plt.subplots()
+    np.random.seed(0)
+    data = np.random.randn(5000)
+    line0, = kde_plot_1d(ax, data, q='5sigma', ncompress=5000)
+    line1, = kde_plot_1d(ax, data, q='5sigma', ncompress=1000)
+    x2y = interp1d(line0.get_xdata(), line0.get_ydata())
+    assert_allclose(line1.get_ydata(), x2y(line1.get_xdata()))
+    plt.close("all")
```
Comment (lukashergt) [inline tests/test_plot.py:341]: This tests the effect of `ncompress` and works nicely for the 1d case, but I can't quite find an equivalent test for the 2d case, where we don't plot the full data, but only the iso contour lines...
Do we need an `ncompress` test for the 2d case? (We didn't have one before...)

**Review by williamjameshandley — DISMISSED** [review]
OK, to get the codecov tests to path I've taken the liberty of putting us closer to 100% coverage -- we'll see how the tests do, but if you're satisfied with this still, please squash and merge!

File: anesthetic/plot.py (hunk 1 above)
Comment (williamjameshandley) [reply anesthetic/plot.py:826]: I think with the new setup whatever density problem this was trying to resolve is now resolved, and it certainly seems to be doing a better job with fewer lines

Comment (lukashergt) [reply anesthetic/plot.py:826]: Actually, these lines _do_ serve a purpose as the following example shows:

```python
data = np.random.randn(10000, 2)
data[:, 1] = data[:, 0]**2
ns = NestedSamples(data=data, columns=['x', 'y'])

fig = plt.figure()
fig, axes = make_2d_axes(['x', 'y'], fig=fig)
fig, axes = ns.plot_2d(axes)
```

### Without those lines:
image


### With those lines:
image

Comment (williamjameshandley) [reply anesthetic/plot.py:826]: Best reinstate them then unless you have another idea.

**Review by williamjameshandley — APPROVED** [review]

**Comment by lukashergt** [comment]
Replaced by #216.

**Pull request closed without merging**